pkg/component/logging/vali/vali.go

```diff
@@ -0,0 +1,927 @@
+						SubPath:   telegrafDataKeyConfig,
+						ReadOnly:  true,
+					},
+					{
+						Name:      telegrafVolumeName,
+						MountPath: telegrafVolumeMountPath + "/" + telegrafDataKeyStartScript,
+						SubPath:   telegrafDataKeyStartScript,
+						ReadOnly:  true,
+					},
+				},
+			},
+		)
+		statefulSet.Spec.Template.Spec.Volumes = append(statefulSet.Spec.Template.Spec.Volumes, corev1.Volume{
+			Name: telegrafVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: telegrafConfigMapName,
+					},
+				},
+			},
+		})
+
+		// TODO: Move `pkg/component/logging/kuberbacproxy` resources into this component.
+		utilruntime.Must(gardenerutils.InjectGenericKubeconfig(statefulSet, genericTokenKubeconfigSecretName, "shoot-access-"+kubeRBACProxyName))
+	}
+
+	utilruntime.Must(references.InjectAnnotations(statefulSet))
+	return statefulSet
+}
+
+func getLabels() map[string]string {
+	return map[string]string{
+		v1beta1constants.GardenRole: v1beta1constants.GardenRoleLogging,
+		v1beta1constants.LabelRole:  "logging",
+		v1beta1constants.LabelApp:   valiName,
+	}
+}
+
+// resizeOrDeleteValiDataVolumeIfStorageNotTheSame updates the Vali PVC if passed storage value is not the same as the
+// current one.
+// Caution: If the passed storage capacity is less than the current one the existing PVC and its PV will be deleted.
+func (v *vali) resizeOrDeleteValiDataVolumeIfStorageNotTheSame(ctx context.Context) error {
+	managedResource := &resourcesv1alpha1.ManagedResource{ObjectMeta: metav1.ObjectMeta{Name: ManagedResourceControlName, Namespace: v.namespace}}
+	removeIgnoreAnnotationFromManagedResource := func() error {
+		// In order to not create the managed resource here first check if exists.
+		if err := v.client.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource); err != nil {
+			if !apierrors.IsNotFound(err) {
+				return err
+			}
+			return nil
+		}
+		_, err := controllerutils.GetAndCreateOrMergePatch(ctx, v.client, managedResource, func() error {
+			delete(managedResource.Annotations, resourcesv1alpha1.Ignore)
+			return nil
+		})
+		return err
+	}
+
+	pvc := &corev1.PersistentVolumeClaim{}
+	if err := v.client.Get(ctx, kubernetesutils.Key(v.namespace, "vali-vali-0"), pvc); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+		return removeIgnoreAnnotationFromManagedResource()
+	}
+
+	// Check if we need resizing
+	storageCmpResult := v.values.Storage.Cmp(*pvc.Spec.Resources.Requests.Storage())
+	if storageCmpResult == 0 {
+		return removeIgnoreAnnotationFromManagedResource()
+	}
+
+	// Annotate managed resource to skip reconciliation.
+	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, v.client, managedResource, func() error {
+		metav1.SetMetaDataAnnotation(&managedResource.ObjectMeta, resourcesv1alpha1.Ignore, "true")
+		return nil
+	}); err != nil {
+		return err
+	}
```

Comment:
We probably shouldn't create it here either when it does not exist, or?
Additionally, maybe you can then use a `Patch` for adding the annotation like suggest above as well.